src/sql/workbench/services/tableDesigner/browser/tableDesignerService.ts

```diff
@@ -11,16 +12,38 @@ import { TableDesignerInput } from 'sql/workbench/browser/editor/tableDesigner/t
 import { IInstantiationService } from 'vs/platform/instantiation/common/instantiation';
 import { IAdsTelemetryService, ITelemetryEventProperties } from 'sql/platform/telemetry/common/telemetry';
 import { TelemetryAction, TelemetryView } from 'sql/platform/telemetry/common/telemetryKeys';
+import { IDialogService } from 'vs/platform/dialogs/common/dialogs';
+import { ILifecycleService } from 'vs/workbench/services/lifecycle/common/lifecycle';
 
 export class TableDesignerService implements ITableDesignerService {
 
 	constructor(@IEditorService private _editorService: IEditorService,
 		@IInstantiationService private _instantiationService: IInstantiationService,
-		@IAdsTelemetryService private _adsTelemetryService: IAdsTelemetryService) { }
+		@IAdsTelemetryService private _adsTelemetryService: IAdsTelemetryService,
+		@ILifecycleService private _lifecycleService: ILifecycleService,
+		@IDialogService private _dialogService: IDialogService) {
+		this._lifecycleService.onBeforeShutdown(async (event) => {
+			event.veto(this.confirmBeforeExit(), 'veto.tableDesigner');
+		});
+	}
 
 	public _serviceBrand: undefined;
 	private _providers = new Map<string, TableDesignerProvider>();
 
+	private async confirmBeforeExit(): Promise<boolean> {
+		let openTableDesignerEditors = this._editorService.editors.filter(e => e instanceof TableDesignerInput);
```

Comment:
seems like we can combine 34 and 35 into one line.

```
		const dirtyTableDesigners = this._editorService.editors.filter(e => e instanceof TableDesignerInput && e.isDirty());
		if (dirtyTableDesigners.length > 0) {
```

Reply:
is there a reason that this comment was not taken?

Reply:
it wasn't a big change and a suggestion so I didn't want to spend time making a minute non functional change and wait for the build.

Reply:
If you particularly want it in your specific style, I'll just edit this in my next PR

Reply:
that would be good. BTW, it would be nice not to leave open comments when you merge the PR, it is ok if you don't want to make certain changes, reply to it and mark it as resolved.